google/cloud/spanner_v1/session.py

```diff
@@ -458,47 +463,100 @@ def run_in_transaction(self, func, *args, **kw):
         )
         attempts = 0
 
-        while True:
-            if self._transaction is None:
-                txn = self.transaction()
-                txn.transaction_tag = transaction_tag
-                txn.exclude_txn_from_change_streams = exclude_txn_from_change_streams
-            else:
-                txn = self._transaction
+        observability_options = getattr(self._database, "observability_options", None)
+        with trace_call(
+            "CloudSpanner.Session.run_in_transaction",
+            self,
+            observability_options=observability_options,
+        ) as span:
+            while True:
+                if self._transaction is None:
+                    add_span_event(span, "Creating Transaction")
+                    txn = self.transaction()
+                    txn.transaction_tag = transaction_tag
+                    txn.exclude_txn_from_change_streams = (
+                        exclude_txn_from_change_streams
+                    )
+                else:
+                    txn = self._transaction
 
-            try:
                 attempts += 1
-                return_value = func(txn, *args, **kw)
-            except Aborted as exc:
-                del self._transaction
-                _delay_until_retry(exc, deadline, attempts)
-                continue
-            except GoogleAPICallError:
-                del self._transaction
-                raise
-            except Exception:
-                txn.rollback()
-                raise
-
-            try:
-                txn.commit(
-                    return_commit_stats=self._database.log_commit_stats,
-                    request_options=commit_request_options,
-                    max_commit_delay=max_commit_delay,
-                )
-            except Aborted as exc:
-                del self._transaction
-                _delay_until_retry(exc, deadline, attempts)
-            except GoogleAPICallError:
-                del self._transaction
-                raise
-            else:
-                if self._database.log_commit_stats and txn.commit_stats:
-                    self._database.logger.info(
-                        "CommitStats: {}".format(txn.commit_stats),
-                        extra={"commit_stats": txn.commit_stats},
+                span_attributes = {"attempt": attempts}
+                txn_id = getattr(txn, "_transaction_id", "") or ""
+                if txn_id:
+                    span_attributes["transaction.id"] = txn_id
+
+                add_span_event(span, "Using Transaction", span_attributes)
```

Comment:
In this part of code there would never be a transaction id available because of inline begin that happens with statement in user operation or with commit.
So there is not point of tracking this here. Lets remove this entire block of code.

Reply:
Alright, thank you @harshachinta! I have removed it.